### ISSUE TITLE
Config flow checks the old configuration

### ISSUE THREAD

**Issue by Ernst79** [custom-components/ble_monitor#656]
It looks like the config flow is doing a check of the old configuration. When I change something in the configuration options, it can cause an error, like in the example below. This needs to be fixed. 

> From the UI I selected "Don't use Bluetooth adapter" and pressed "Submit", but a red error message was shown on the top (User input malformed: 00:00:00:00:00:00 is not a valid option for dictionary value @ `data['bt_interface']`) and the configuration window stays open.
> 
> I changed it manually in the `core.config_entries` file in HA and then everything worked fine:
> 

```
            {
                "entry_id": "d30216eefbc6f8d33468b6c3e444dda7",
                "version": 3,
                "domain": "ble_monitor",
                "title": "Bluetooth Low Energy Monitor",
                "data": {},
                "options": {
                    "bt_interface": [
                       "00:00:00:00:00:00" -> "disable" # changed it here manually to disable and then after HA restart it worked as expected
                    ],
                    "bt_auto_restart": false,
                    "active_scan": false,
                    "discovery": false,#
                   ...
```

**Comment by Ernst79** [comment]
Created a "temporary" fix in 7.3.4-beta, if someone has `"bt_interface": ["00:00:00:00:00:00"]`, it will be updated to `"bt_interface": ["disable"]` automatically. 

I'll keep this issue open, as the same applies for the situation where someone moves his config to a different machine, by applying a backup. In that case, the old MAC is wrong, which results in the same error.

**Comment by Ernst79** [comment]
Fixed in 7.3.5-beta

**Comment by Ernst79** [comment]
Fixed in 7.3.8
